hls4ml/model/hls_layers.py

```diff
@@ -1529,7 +1533,7 @@ def initialize(self):
 
     def function_cpp(self):
         params = self._default_function_params()
-        params['dim'] = self.get_attr('dim')
+        params['dim'] = '3d' # always use 3d instead of self.get_attr('dim')
```

Comment:
I think we should reevaluate this decision and consider making a 2d transpose as well. It would be simpler and compile-time constant, therefore better for Vivado than the general 3d one that is difficult for the compiler to optimize.

Reply:
Yes, I think it makes sense to make a new 2d specific transpose